numpy/random/setup.py

```diff
@@ -40,23 +44,119 @@ def generate_libraries(ext, build_dir):
         defs.append(("NPY_NEEDS_MINGW_TIME_WORKAROUND", None))
 
     libs = []
-    # Configure mtrand
-    config.add_extension('mtrand',
-                         sources=[join('mtrand', x) for x in
-                                  ['mtrand.c', 'randomkit.c', 'initarray.c',
-                                   'distributions.c']]+[generate_libraries],
-                         libraries=libs,
-                         depends=[join('mtrand', '*.h'),
-                                  join('mtrand', '*.pyx'),
-                                  join('mtrand', '*.pxi'),],
-                         define_macros=defs,
-                         )
-
-    config.add_data_files(('.', join('mtrand', 'randomkit.h')))
+    defs.append(('NPY_NO_DEPRECATED_API', 0))
     config.add_data_dir('tests')
 
-    return config
+    ##############################
+    # randomgen
+    ##############################
+
+    # Make a guess as to whether SSE2 is present for now, TODO: Improve
+    DEBUG = False
+    EXTRA_LINK_ARGS = []
+    EXTRA_LIBRARIES = ['m'] if os.name != 'nt' else []
+    EXTRA_COMPILE_ARGS = [] if os.name == 'nt' else [
+        '-std=c99', '-U__GNUC_GNU_INLINE__']
+    if os.name == 'nt':
+        EXTRA_LINK_ARGS = ['/LTCG', '/OPT:REF', 'Advapi32.lib', 'Kernel32.lib']
```

Comment:
This is the problematic block.  I think it needs to look more like 

https://github.com/numpy/numpy/blob/master/numpy/random/setup.py#L27

Reply:
I don't have a mingw setup. Can you test if adding `mingw32` like 
```
from numpy.distutils.misc_utils import mingw32
...
if os.name == 'nt' and not mingw32():
```
solves it?

Reply:
I made a PR that cleans this up a lot and I think should work on many untested platform/compiler combinations.